File: application/src/main/java/de/wuespace/telestion/project/daedalus2/redis/RedisTimeSeriesSpecification.java
package de.wuespace.telestion.project.daedalus2.redis;

import com.fasterxml.jackson.annotation.JsonProperty;
import de.wuespace.telestion.api.message.JsonMessage;

import java.util.List;

@SuppressWarnings("unused")
public record RedisTimeSeriesSpecification(
		@JsonProperty String key,
		@JsonProperty String from,
		@JsonProperty String to,
		@JsonProperty int bucketSize,
		@JsonProperty List<String> aggregations
) implements JsonMessage {
	public RedisTimeSeriesSpecification() {
		this(null, null, null, 500, null);
	}
}

File: client/src/widgets/hooks/redis/model/redis-time-series-specification.ts
import { AggregationType } from '../use-request-time-series';
import { JavaMessage } from '../../../../model/java-message';

/**
 * Counterpart to `de.wuespace.telestion.project.daedalus2.redis.RedisTimeSeriesSpecification` Java class
 */
export interface RedisTimeSeriesSpecification extends JavaMessage {
	key: string;
	from: string;
	to: string;
	bucketSize: number;
	aggregations: AggregationType[];
	className: 'de.wuespace.telestion.project.daedalus2.redis.RedisTimeSeriesSpecification';
}
